### PR TITLE
kernel_interface: check fs existing or not with correct target

### PR DESCRIPTION
**Pull request by yanan-fu**
1. kernel_interface: Avoid repeated calls to 'fs_value'
2. kernel_interface: check fs existing or not with correct target

id: 1871870
Signed-off-by: Yanan Fu <yfu@redhat.com>

### CODE REVIEW

**Comment by yanan-fu** [comment]
@vivianQizhu  Could you help review ?
I move the fix of repeated call from https://github.com/avocado-framework/avocado-vt/pull/2699 to this one, since i want to fix the second issue "check fs existing or not with correct target", they need modify the same file.

Leave https://github.com/avocado-framework/avocado-vt/pull/2699 to fix the NumaInfo issue only. Thanks a lot!

**Comment by yanan-fu** [comment]
The CI failing is not related with the fix:
```
avocado list: error: unrecognized arguments: --vt-type=libvirt
Perhaps a plugin is missing; run 'avocado plugins' to list the installed ones
```

Will wait and force update again to see if it will gone.

**Review by sathnaga — APPROVED** [review]
LGTM

**Comment by sathnaga** [comment]
```
>>> from virttest import kernel_interface
>>> testobj = kernel_interface.ProcFS("/home/sath/tests/avocado-vt/test")
>>> testobj.proc_fs_value
'testing'
>>> testobj.proc_fs_value=1
>>> testobj.proc_fs_value
1
>>>
```

**Comment by yanan-fu** [comment]
@sathnaga Thanks, i update the patch to fix the cyclic import issue since:
https://github.com/avocado-framework/avocado-vt/blob/2cc74315c7eb4f05d052428e79a4f4739754a601/virttest/utils_misc.py#L85

 Could you help review again ? Thanks!

**Comment by vivianQizhu** [comment]
@yanan-fu Why not make it a class member so that you don't need to pass the session object?

**Comment by yanan-fu** [comment]
> @yanan-fu Why not make it a class member so that you don't need to pass the session object?

Same for filename(self.fs) ?  Then no need pass any param, do you mean this ?

**Comment by vivianQizhu** [comment]
@yanan-fu Yes.

**Comment by yanan-fu** [comment]
> @yanan-fu Yes.

Updated, could you help review again, thanks~

File: virttest/kernel_interface.py [hunk 1]
```diff
@@ -17,13 +17,23 @@ def __init__(self, fs, session=None):
...
         self.session = session
+        if not self._check_isfile():
+            raise AttributeError("%s is not available" % self.fs)
         if self.session:
             self.func = self.session.cmd_status_output
 
+    def _check_isfile(self):
+        """
+        check whether the fs exist in local/remote host/VM
+
```
Comment (vivianQizhu) [inline virttest/kernel_interface.py:30]: This blank line might be removed.

Comment (yanan-fu) [reply virttest/kernel_interface.py]: > This blank line might be removed.

Ah, yes, delete the blank line now, thanks!

**Review by vivianQizhu — APPROVED** [review]

**Pull request merged**